qualification/report/generate_pdf.py

```diff
@@ -835,14 +826,64 @@ def document_from_json(input_data: str | list) -> Document:
     return doc
 
 
+def test_image_commit(result_list: list) -> str:
+    """Extract the git commit ID of the test image from the test results.
+
+    Parameters
+    ----------
+    result_list: list
+        A list containing the test results.
+
+    Returns
+    -------
+    commit_id: str
+        The git commit ID of the test image.
+    """
+    test_configuration = parse(result_list)[0]
+    git_versions = set()
+    for correlator in test_configuration.correlators:
+        for task_name, task in correlator.tasks.items():
+            if task_name != "product_controller":
+                git_versions.add(task.git_version)
+    return collapse_versions(git_versions)
+
+
+def list_from_json(input_data: str | list) -> list:
+    """Take a test result and generate a list.
+
+    Parameters
+    ----------
+    input_data
+        Either the list of parsed JSON entries, or a path to the report file
+        generated by :samp:`pytest --report-log={filename}`.
```

Comment:
You can remove the support for taking a list as input. In that case it's just returning what it's given, and I don't see a use case for that.

Reply:
Doesn't seem to be resolved.

Reply:
Apologies, I missed the usage of the list in the code.